staging/src/k8s.io/client-go/util/cert/cert.go

```diff
@@ -138,6 +138,34 @@ func MakeEllipticPrivateKeyPEM() ([]byte, error) {
 // Host may be an IP or a DNS name
 // You may also specify additional subject alt names (either ip or dns names) for the certificate
 func GenerateSelfSignedCertKey(host string, alternateIPs []net.IP, alternateDNS []string) ([]byte, []byte, error) {
+	caKey, err := rsa.GenerateKey(cryptorand.Reader, 2048)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	caTemplate := x509.Certificate{
+		SerialNumber: big.NewInt(1),
+		Subject: pkix.Name{
+			CommonName: fmt.Sprintf("%s-ca@%d", host, time.Now().Unix()),
+		},
+		NotBefore: time.Now(),
+		NotAfter:  time.Now().Add(time.Hour * 24 * 365),
+
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
```

Comment:
At least in kops we use `	KeyUsage:              x509.KeyUsageCertSign | x509.KeyUsageCRLSign,`, and I think GKE just has:

```
        X509v3 extensions:
            X509v3 Key Usage: critical
                Certificate Sign
            X509v3 Basic Constraints: critical
                CA:TRUE
```

Not sure if it matters...

Reply:
was copied from here:
https://github.com/liggitt/kubernetes/blob/7488d1c9210e60aef9ad49f07cb5d8a24152db88/staging/src/k8s.io/client-go/util/cert/cert.go#L62-L76

additional usages should be fine